src/ui/ui.ts

```diff
@@ -128,79 +173,206 @@ export class UI extends Phaser.GameObjects.Container {
+      case UiMode.CONFIRM:
+        return new ConfirmUiHandler();
+      case UiMode.OPTION_SELECT:
+        return new OptionSelectUiHandler();
+      case UiMode.MENU:
+        return new MenuUiHandler();
+      case UiMode.MENU_OPTION_SELECT:
+        return new OptionSelectUiHandler(UiMode.MENU_OPTION_SELECT);
+      // settings
+      case UiMode.SETTINGS:
+        return new GeneralSettingsUiHandler();
+      case UiMode.SETTINGS_DISPLAY:
+        return new DisplaySettingsUiHandler();
+      case UiMode.SETTINGS_AUDIO:
+        return new AudioSettingsUiHandler();
+      case UiMode.SETTINGS_GAMEPAD:
+        return new GamepadSettingsUiHandler();
+      case UiMode.GAMEPAD_BINDING:
+        return new GamepadBindingUiHandler();
+      case UiMode.SETTINGS_KEYBOARD:
+        return new KeyboardSettingsUiHandler();
+      case UiMode.KEYBOARD_BINDING:
+        return new KeyboardBindingUiHandler();
+      case UiMode.ACHIEVEMENTS:
+        return new AchievementsUiHandler();
+      case UiMode.GAME_STATS:
+        return new GameStatsUiHandler();
+      case UiMode.EGG_LIST:
+        return new EggListUiHandler();
+      case UiMode.EGG_GACHA:
+        return new EggGachaUiHandler();
+      case UiMode.LOGIN_FORM:
+        return new LoginFormUiHandler();
+      case UiMode.REGISTRATION_FORM:
+        return new RegistrationFormUiHandler();
+      case UiMode.LOADING:
+        return new LoadingModalUiHandler();
+      case UiMode.SESSION_RELOAD:
+        return new SessionReloadModalUiHandler();
+      case UiMode.UNAVAILABLE:
+        return new UnavailableModalUiHandler();
+      case UiMode.CHALLENGE_SELECT:
+        return new ChallengeSelectUiHandler();
+      case UiMode.RENAME_POKEMON:
+        return new RenamePokemonUiHandler();
+      case UiMode.RUN_HISTORY:
+        return new RunHistoryUiHandler();
+      case UiMode.RUN_INFO:
+        return new RunInfoUiHandler();
+      case UiMode.TEST_DIALOGUE:
+        return new TestDialogueUiHandler();
+      case UiMode.AUTO_COMPLETE:
+        return new AutoCompleteUiHandler();
+      case UiMode.ADMIN:
+        return new AdminUiHandler();
+      case UiMode.MYSTERY_ENCOUNTER:
+        return new MysteryEncounterUiHandler();
+    }
+  }
+
+  /**
+   * Initialize a new UiHandler for the given mode, and add it the handlers list.
+   * @param mode - The {@linkcode UiMode} to a create handler for
+   * @returns a ready to use {@linkcode UiHandler} for that mode
+   */
+  private addUiHandler(mode: UiMode): UiHandler {
+    const handler = this.createUiHandler(mode);
+    logUiEvent(`Initializing Handler for mode: ${UiMode[mode]} (${mode})`);
+    this.handlers.set(mode, handler);
+    handler.initialize();
+    return handler;
+  }
+
+  /**
+   * Destroy the handlers for the given modes if they exist, and remove them from the handlers list.
+   * @param modes - one or more {@linkcode UiMode}s
+   */
+  private deleteUiHandlers(modes: UiMode | UiMode[]): void {
```

Comment:
We should consider making this a `vararg` instead.

```suggestion
  private deleteUiHandlers(...modes: UiMode[]): void {
```

Reply:
Agreed, in fact it's probably a good idea to go through the codebase and find more places this can be done easily (this pattern is most of the cause of the need for the `coerceArray` utility function iirc). Will add an issue for it.

Edit: Issue added #1256